crates/typst/src/layout/grid.rs

```diff
@@ -278,727 +267,119 @@ cast! {
     values: Array => Self(values.into_iter().map(Value::cast).collect::<StrResult<_>>()?),
 }
 
-/// A value that can be configured per cell.
-#[derive(Debug, Clone, PartialEq, Hash)]
-pub enum Celled<T> {
-    /// A bare value, the same for all cells.
-    Value(T),
-    /// A closure mapping from cell coordinates to a value.
-    Func(Func),
-    /// An array of alignment values corresponding to each column.
-    Array(Vec<T>),
-}
+/// A cell in the grid. Use this to either override grid properties for a
+/// particular cell, or in show rules to apply certain styles to multiple cells
+/// at once.
+///
+/// For example, you can override the fill, alignment or inset for a single
+/// cell:
+///
+/// ```example
+/// #grid(
+///   columns: 2,
+///   fill: red,
+///   align: left,
+///   inset: 5pt,
+///   [ABC], [ABC],
+///   grid.cell(fill: blue)[C], [D],
+///   grid.cell(align: center)[E], [F],
+///   [G], grid.cell(inset: 0pt)[H]
+/// )
+/// ```
+///
+/// You can also use show rules to style all cells at once:
+///
+/// ```example
+/// #{
+///   // Creating a temporary #{ ... } scope to avoid affecting other grids
+///   // with this show rule.
+///   // Here, we will italicize all cells.
+///   show grid.cell: emph
```

Comment:
I'm not sure, I think with the PR at the current state we'd have to add a fairly non-standard example here (e.g. apply `emph` based on alignment or something). With the `x`, `y` fields PR, however, we'd probably have more to show, since we could e.g. make only the first row bold.

Reply:
Then maybe let's remove the example until we have something idiomatic to show. Or I guess keep and change it later. I don't mind either way.

Reply:
Sure, removing sounds fair for now.